src/runtime/crt/memory/stack_allocator.c

```diff
@@ -79,8 +79,13 @@ tvm_crt_error_t StackMemoryManager_Free(tvm_workspace_t* tvm_runtime_workspace,
 
 tvm_crt_error_t StackMemoryManager_Init(tvm_workspace_t* tvm_runtime_workspace,
                                         uint8_t* g_aot_memory, size_t workspace_size) {
-  tvm_runtime_workspace->next_alloc = g_aot_memory;
-  tvm_runtime_workspace->workspace = g_aot_memory;
-  tvm_runtime_workspace->workspace_size = workspace_size;
+  // Calculate g_aot_memory misalignment and offset
+  uint32_t extra = (uintptr_t)g_aot_memory % TVM_RUNTIME_ALLOC_ALIGNMENT_BYTES;
```

Comment:
i think you could simplify this as:
```
uintptr_t unaligned_mask = TVM_RUNTIME_ALLOC_ALIGNMENT_BYTES - 1;
g_aot_memory = ((uintptr_t) g_aot_memory) + unaligned_mask) & ~unaligned_mask;
```

also can you update the comment to explain we need to round up g_aot_memory in case it is not aligned to `TVM_RUNTIME_ALLOC_ALIGNMENT_BYTES`?

Reply:
done.